src/bridge/cockpitrouter.c

```diff
@@ -77,12 +156,30 @@ on_channel_closed (CockpitChannel *channel,
                    gpointer user_data)
 {
   CockpitRouter *self = user_data;
+  ExternalBridge *bridge = NULL;
   const gchar *id;
 
   id = cockpit_channel_get_id (channel);
   g_hash_table_remove (self->channels, id);
   g_hash_table_remove (self->groups, id);
 
+  bridge = g_hash_table_lookup (self->bridges_by_channel, id);
+  if (bridge)
+    {
+      g_hash_table_remove (self->bridges_by_channel, id);
+      g_hash_table_remove (bridge->channels, id);
+      if (g_hash_table_size (bridge->channels) == 0)
+        {
+          /*
+           * Close sessions that are no longer in use after N seconds
+           * of them being that way.
```

Comment:
You're right we probably need a way to turn this off for some bridges, can that be a follow up though?